crepecake/src/main/java/com/google/cloud/tools/crepecake/http/Connection.java

```diff
@@ -40,10 +42,15 @@
  */
 public class Connection implements Closeable {
 
-  private static final HttpRequestFactory HTTP_REQUEST_FACTORY =
-      new NetHttpTransport().createRequestFactory();
-
-  private HttpRequestFactory requestFactory = HTTP_REQUEST_FACTORY;
+  /**
+   * Do not use {@link NetHttpTransport}. It does not process response errors properly. A new {@link
+   * ApacheHttpTransport} needs to be created for each connection because otherwise HTTP connection
+   * persistence causes the connection to throw {@link NoHttpResponseException}.
+   *
+   * @see <a
```

Comment:
Is this a bug, or working as intended?

Reply:
This is a working as intended for the library but a breaking feature for us. Essentially, NetHttpTransport adheres strictly to the RFC specs and breaks if it processes messages that don't fit the specs. Essentially, in this case, it throws an IOException on `401 Unauthorized` responses that don't have a `WWW-Authenticate` header, which is the case for certain Docker registry messages. I believe there are other Docker registry messages also don't follow the RFC specs exactly.